backend/src/controller/modelController.ts

```diff
@@ -37,32 +36,23 @@ function handleConfigureModel(req: OpenAiConfigureModelRequest, res: Response) {
 	const configId = req.body.configId as MODEL_CONFIG | undefined;
 	const value = req.body.value;
 
-	let updated = null;
-
-	if (configId && value && value >= 0) {
+	if (configId && value !== undefined) {
 		const lastConfig = req.session.chatModel.configuration;
-		switch (configId) {
-			case MODEL_CONFIG.TEMPERATURE:
-				updated = updateConfigProperty(lastConfig, configId, value, 2);
-				break;
-			case MODEL_CONFIG.TOP_P:
-				updated = updateConfigProperty(lastConfig, configId, value, 1);
-				break;
-			case MODEL_CONFIG.FREQUENCY_PENALTY:
-				updated = updateConfigProperty(lastConfig, configId, value, 2);
-				break;
-			case MODEL_CONFIG.PRESENCE_PENALTY:
-				updated = updateConfigProperty(lastConfig, configId, value, 2);
-				break;
-			default:
-				res.status(400).send();
-		}
-		if (updated) {
-			req.session.chatModel.configuration = updated;
+		const maxValue = configId === MODEL_CONFIG.TOP_P ? 1 : 2;
```

Comment:
I'd personally be tempted to move this above the `if` statement on line 39, and then add a range check to that `if` statement. 
Then, you don't even need the `updateConfigProperty` method anymore, just do 

```typescript
function handleConfigureModel(req: OpenAiConfigureModelRequest, res: Response) {
	const configId = req.body.configId as MODEL_CONFIG | undefined;
	const value = req.body.value;
	const maxValue = configId === MODEL_CONFIG.TOP_P ? 1 : 2;

	if (configId && value !== undefined && value >= 0 && value <= maxValue) {
		req.session.chatModel.configuration[configId] = value;
		res.status(200).send();
	} else {
		res.status(400).send();
	}
}
```

Reply:
ah yeah much nicer!